common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -754,7 +753,36 @@ private static String statusMsg (String status){
             return ", with the status : " + status;
         }
 
+    /***
+     * Stops the SilentRequestsExecutor.
+     * Waits 1 Sec for existing silent requests to finish, before terminating them.
+     * WARN!! No new silent requests will be processed after this until the executor is reset
+     * This is expected to be called when in Shared Device Mode when global signout is performed.
+     */
+    public static void stopSilentRequestExecutor() {
+        final String methodTag = TAG + ":stopSilentRequestExecutor";
+        Logger.info(methodTag, "shutting down..");
+        sSilentExecutor.shutdown();
+        try {
+            if (!sSilentExecutor.awaitTermination(1000, TimeUnit.MILLISECONDS)) {
+                Logger.warn(methodTag, "terminating now");
+                sSilentExecutor.shutdownNow();
+            }
+        } catch (final InterruptedException e) {
+            Logger.warn(methodTag, "terminating again");
+            sSilentExecutor.shutdownNow();
+        }
+    }
 
-
+    /***
+     * Resets the SilentRequestsExecutor.
+     * This creates a new Executor for the silent request.
+     * This is expected to be called after global signout is performed in Shared Device mode.
+     * This should be called if previously the Executor was stopped using 'stopSilentRequestExecutor'
+     */
+    public static void resetSilentRequestExecutor() {
+        Logger.info(TAG + ":resetSilentRequestExecutor", "Resetting silent Executor");
```

Comment:
What if the executor was reset without calling stopSilentRequestExecutor prior?

Reply:
It will create a new pool to start executing silent requests.